.github/workflows/build-test.yml

```diff
@@ -114,11 +157,25 @@ jobs:
             - name: Install Homebrew Bash
               shell: bash
               run: brew install bash
+            - name: Restore cached deps
+              id: cache-restore-deps
+              uses: actions/cache/restore@v3
+              with:
+                path: dep-cache
+                key: ${{ matrix.build_type }}-${{ github.event.pull_request.base.sha }}-Mac-dep-cache
+              if: github.event_name == 'pull_request'
             - name: Build
               shell: bash
               env:
                 BUILD_TYPE: ${{ matrix.build_type }}
               run: ./ci/build_mac.sh
+            - name: Cache deps
+              id: cache-save-deps
+              uses: actions/cache/save@v3
+              with:
+                path: dep-cache
+                key: ${{ matrix.build_type }}-${{ github.sha }}-Mac-dep-cache
+              if: github.ref_name == github.event.repository.default_branch && matrix.cloud_provider == 'AWS'
```

Comment:
we save to different place that we restore from - is it correct?

Reply:
We build and cache deps on master (commit sha = github.sha, but we restore on pull requests on master. On pull requests we will restore deps based on a base_ref - some commit on master (commit sha = github.event.pull_request.base.sha). So we restore and save from the same cache -> just not on the same workflow.